### ISSUE TITLE
Can't get prospector to show output

### ISSUE THREAD

**Issue by chocoelho** [vim-syntastic/syntastic#2024]
I followed the instructions at #1315, and got this output:

```
"<path>" 89L, 3088C                                                                                                                                                                                  
Syntastic version: 3.8.0-55 (Vim 704, Neovim, Linux)                                                                                                                                                                                        
Info for filetype: python                                                                                                                                                                                                                   
Global mode: active                                                                                                                                                                                                                         
Filetype python is active                                                                                                                                                                                                                   
The current file will be checked automatically                                                                                                                                                                                              
syntastic: 23.804705: system: command run in 0.526915s                                                                                                                                                                                      
syntastic: 23.805081: python/pylint: pylint version = [1, 7, 1]                                                                                                                                                                             
syntastic: 23.809334: system: command run in 0.002840s                                                                                                                                                                                      
syntastic: 23.809621: python/python: getVersion: 'python --version': ['Python 3.6.1', '']                                                                                                                                                   
syntastic: 23.809946: python/python: python version = [3, 6, 1]                                                                                                                                                                             
Available checkers: bandit flake8 mypy pep257 pep8 prospector pycodestyle pydocstyle pyflakes pylint python                                                                                                                                 
Currently enabled checkers: prospector mypy                                                                                                                                                                                                 
syntastic: 23.811855: &shell = '/bin/sh' (!), &shellcmdflag = '-c', &shellpipe = '2>&1| tee', &shellquote = '', &shellredir = '>%s 2>&1', &shelltemp = 1, &shellxquote = '', &autochdir = 0, &shellxescape = ''                             
syntastic: 36.392812: g:syntastic_version = '3.8.0-55 (Vim 704, Neovim, Linux)'                                                                                                                                                             
syntastic: 36.393737: &shell = '/bin/sh' (!), &shellcmdflag = '-c', &shellpipe = '2>&1| tee', &shellquote = '', &shellredir = '>%s 2>&1', &shelltemp = 1, &shellxquote = '', &autochdir = 0, &shellxescape = ''                             
syntastic: 36.395983: UpdateErrors: prospector                                                                                                                                                                                              
syntastic: 36.397452: CacheErrors: prospector                                                                                                                                                                                               
syntastic: 36.400595: g:syntastic_aggregate_errors = 0                                                                                                                                                                                      
syntastic: 36.401494: $TERM = 'xterm'                                                                                                                                                                                                       
syntastic: 36.402136: $TMPDIR = ''                                                                                                                                                                                                          
syntastic: 36.402616: $PATH = '/home/chocoelho/.virtualenvs/<venv>i/bin:/home/chocoelho/.pyenv/plugins/pyenv-virtualenvwrapper/shims:/home/chocoelho/.pyenv/libexec:/home/chocoelho/.pyenv/plugins/python-build/bin:/home/chocoelho/.py
env/plugins/pyenv-virtualenvwrapper/bin:/home/chocoelho/.pyenv/plugins/pyenv-update/bin:/home/chocoelho/.pyenv/plugins/pyenv-installer/bin:/home/chocoelho/.pyenv/plugins/pyenv-doctor/bin:/home/chocoelho/.pyenv/shims:/home/chocoelho/.pye
nv/bin:/home/chocoelho/bin:/home/chocoelho/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin'                                                                                   
syntastic: 36.406903: getcwd() = '<path>'                                                                                                                                                                                  
syntastic: 36.409630: CacheErrors: Invoking checker: python/prospector                                                                                                                                                                      
syntastic: 36.412629: SyntasticMake: called with options: {'preprocess': 'prospector', 'errorformat': '%f:%l:%c: %m', 'makeprg': 'prospector --messages-only --absolute-paths --die-on-tool-error --zero-exit --output-format json trademark
s/helpers/euipo/dailies.py', 'env': {'TERM': 'dumb'}, 'returns': [0]}                                                                                                                                                                       
syntastic: 38.046198: system: command run in 1.631469s                                                                                                                                                                                      
syntastic: 38.051119: getLocList: checker python/prospector returned 0                                                                                                                                                                      
syntastic: 38.051386: getLocList: checker python/prospector run in 1.641114s 
```

When running the generated command in a separate bash window, it shows something like this:

```
('utf-8', [b'import logging\n'])
{
  "messages": [
    {
      "source": "pep8",
      "code": "E701",
      "location": {
        "path": "<path>",
        "module": null,
        "function": null,
        "line": 9,
        "character": 7
      },
      "message": "multiple statements on one line (colon)"
    },
    ...
```

The thing is: I can't get prospector to show when using it inside NeoVim :(

EDIT: This is the `.prospector.yaml` that I'm using:

```
output-format: text

strictness: veryhigh
test-warnings: true
doc-warnings: false
member-warnings: true

uses:
  - django
  - celery

pep8:
  full: true
  disable:
    - D100
    - D101
    - D102
    - D103
    - D105
    - D205
    - D400 
    - E701
  options:
    max-line-length: 100

pylint:
  options:
    max-line-length: 100

vulture:
  run: true

ignore-paths:
  - node_modules
  - venv
  - env

ignore-patterns:
  - (^|/)migrations(/|$)
  - (^|/)settings(/|$)
```

**Comment by lcd047** [comment]
Please open your test file, set `g:syntastic_debug` to 3, run the checker, run `:mes`, and post the output.

**Comment by chocoelho** [comment]
If set to 3, the output is: 
```
"~/src/api/trademarks/helpers/euipo/dailies.py" 86L, 3012C                                                                                                                                                                                  
syntastic: 15.000012: g:syntastic_version = '3.8.0-55 (Vim 704, Neovim, Linux)'                                                                                                                                                             
syntastic: 15.000693: &shell = '/bin/sh' (!), &shellcmdflag = '-c', &shellpipe = '2>&1| tee', &shellquote = '', &shellredir = '>%s 2>&1', &shelltemp = 1, &shellxquote = '', &autochdir = 0, &shellxescape = ''                             
syntastic: 15.004241: UpdateErrors: prospector                                                                                                                                                                                              
syntastic: 15.005828: CacheErrors: prospector                                                                                                                                                                                               
syntastic: 15.009087: g:syntastic_aggregate_errors = 0                                                                                                                                                                                      
syntastic: 15.010368: getcwd() = '/home/chocoelho/src/api'                                                                                                                                                                                  
syntastic: 15.012799: CacheErrors: Invoking checker: python/prospector                                                                                                                                                                      
syntastic: 15.016537: SyntasticMake: called with options: {'preprocess': 'prospector', 'errorformat': '%f:%l:%c: %m', 'makeprg': 'prospector --messages-only --absolute-paths --die-on-tool-error --zero-exit --output-format json trademark
s/helpers/euipo/dailies.py', 'env': {'TERM': 'dumb'}, 'returns': [0]}                                                                                                                                                                       
syntastic: 17.071480: system: command run in 2.049962s                                                                                                                                                                                      
syntastic: 17.072090: checker output: ['(''utf-8'', [b''import logging\n''])', '{', '  "messages": [', '    {', '      "source": "pyflakes",', '      "code": "F401",', '      "location": {', '        "path": "/home/chocoelho/src/api/tra
demarks/helpers/euipo/dailies.py",', '        "module": null,', '        "function": null,', '        "line": 4,', '        "character": 1', '      },', '      "message": "''typing.Any'' imported but unused"', '    },', '    {', '      
"source": "pylint",', '      "code": "unused-argument",', '      "location": {', '        "path": "/home/chocoelho/src/api/trademarks/helpers/euipo/dailies.py",', '        "module": "trademarks.helpers.euipo.dailies",', '        "functi
on": "euipo_stock_daily",', '        "line": 61,', '        "character": 56', '      },', '      "message": "Unused argument ''stock_mode''"', '    },', '    {', '      "source": "vulture",', '      "code": "unused-function",', '      "
location": {', '        "path": "/home/chocoelho/src/api/trademarks/helpers/euipo/dailies.py",', '        "module": null,', '        "function": null,', '        "line": 22,', '        "character": null', '      },', '      "message": "
Unused function download_euipo_file"', '    },', '    {', '      "source": "vulture",', '      "code": "unused-function",', '      "location": {', '        "path": "/home/chocoelho/src/api/trademarks/helpers/euipo/dailies.py",', '      
  "module": null,', '        "function": null,', '        "line": 80,', '        "character": null', '      },', '      "message": "Unused function fetch_daily_files"', '    }', '  ]', '}', '']                                           
syntastic: 17.075403: preprocess: []                                                                                                                                                                                                        
syntastic: 17.075682: raw loclist: []                                                                                                                                                                                                       
syntastic: 17.075936: getLocList: checker python/prospector returned 0                                                                                                                                                                      
syntastic: 17.076101: python/prospector raw: []                                                                                                                                                                                             
syntastic: 17.076304: quiet_messages filter: {}                                                                                                                                                                                             
syntastic: 17.076443: getLocList: checker python/prospector run in 2.063010s                                                                                                                                                                
syntastic: 17.076754: aggregated: {'_sorted': 0, '_name': '', '_owner': 1, '_columns': 1, '_rawLoclist': []}                                                                                                                                                                            
```

**Comment by lcd047** [comment]
The output of `prospector` starts with `('utf-8', [b'import logging\n'])`.  This is not valid JSON, so feel free to complain to `prospector` developers about it.

On syntastic's side, there probably should be an error message about input being invalid.  I think there was a reason for not adding that particular check at top level, but I can't remember it right now.

**Comment by chocoelho** [comment]
Nice, so I'll take a better look at `prospector`. Thanks for pointing that out.

**Comment by chocoelho** [comment]
In landscapeio/prospector#210 they said that they left some debug statements over there. So just to put it here as a reference, this in on `prospector` side :)

**Comment by carlio** [comment]
Derp. Sorry about that. Just released version 0.12.6 to PyPI with this fixed. It was only introduced in 0.12.5 so just avoid that released.
